csrc/scheduler/utils.cpp

```diff
@@ -764,6 +764,20 @@ PersistentBufferInfo persistentBuffers(Fusion* fusion) {
   return persistent_buffer_info;
 }
 
+namespace {
+int64_t getAllocatedExtent(
+    TensorView* tv,
+    IterDomain* id,
+    ExpressionEvaluator& expr_eval) {
+  IterDomain* alloc_id = projectLogicalToAllocation(tv, id);
+  auto inferred_val = expr_eval.evaluate(alloc_id->extent());
+  NVF_ERROR(
+      inferred_val.hasValue(),
+      "Error inferring dimensions of reduction fusion.");
+  return inferred_val.as<int64_t>();
+}
+} // namespace
+
 ReductionTvProperties getReductionProperties(
```

Comment:
Could you add comments with some examples illustrating how a sharded tensor would work?